Assets/Scripts/TwitchClient.cs

```diff
@@ -0,0 +1,124 @@
+		_client.OnChatCommandReceived += OnChatCommandReceived;
+        _client.OnWhisperReceived += _client_OnWhisperReceived;
+
+		// Connect
+		_client.Connect();
+
+	}
+
+	private void _client_OnWhisperReceived(object sender, TwitchLib.Client.Events.OnWhisperReceivedArgs e)
+	{
+		Debug.Log("Message: " + e.WhisperMessage.Message);
+		
+		_client.SendWhisper(e.WhisperMessage.Username, "Its working!");
+	}
+
+    private void _client_OnError(object sender, TwitchLib.Communication.Events.OnErrorEventArgs e)
+    {
+		Debug.LogError($"The program didnt work correctly: {e.Exception.Message}");
+    }
+
+    private void OnConnected(object sender, TwitchLib.Client.Events.OnConnectedArgs e)
+	{
+		Debug.Log($"The bot {e.BotUsername} succesfully connected to Twitch.");
+
+		if (!string.IsNullOrWhiteSpace(e.AutoJoinChannel))
+			Debug.Log($"The bot will now attempt to automatically join the channel provided when the Initialize method was called: {e.AutoJoinChannel}");
+	}
+    private void OnFailedToConnect(object sender, TwitchLib.Client.Events.OnConnectionErrorArgs e)
+    {
+		Debug.Log($"The bot {e.BotUsername} failed to connect");
+    }
+
+	private void OnJoinedChannel(object sender, TwitchLib.Client.Events.OnJoinedChannelArgs e)
+	{
+		Debug.Log($"The bot {e.BotUsername} just joined the channel: {e.Channel}");
+		_client.SendMessage(e.Channel, "I just joined the channel! PogChamp");
+	}
+
+	private void OnMessageReceived(object sender, TwitchLib.Client.Events.OnMessageReceivedArgs e)
+	{
+		Debug.Log($"Message received from {e.ChatMessage.UserId}: {e.ChatMessage.Message}");
+	}
+
+	
+	private void OnChatCommandReceived(object sender, TwitchLib.Client.Events.OnChatCommandReceivedArgs e)
+	{
+		switch (e.Command.CommandText)
+		{
+			case "d":
+				//_client.SendWhisper(e.Command.ChatMessage.Username, "Its working!");
+				_client.SendMessage(e.Command.ChatMessage.Channel, "You moved the twitch bot to the right");
+				robotMovement.MoveSideways(SidewaysDirections.right);
+				break;
+			case "a":
+				_client.SendMessage(e.Command.ChatMessage.Channel, "You moved the twitch bot to the left");
+				robotMovement.MoveSideways(SidewaysDirections.left);
+				break;
+			case "w":
+				_client.SendMessage(e.Command.ChatMessage.Channel, "You moved the twitch bot up");
+				robotMovement.MoveSideways(SidewaysDirections.up);
+				break;
+			case "s":
+				_client.SendMessage(e.Command.ChatMessage.Channel, "You moved the twitch bot down");
+				robotMovement.MoveSideways(SidewaysDirections.down);
+				break;
+			default:
+				_client.SendMessage(e.Command.ChatMessage.Channel, $"Unknown chat command: {e.Command.CommandIdentifier}{e.Command.CommandText}");
+				break;
+		}
+	}
+
+
+
+    private void Update()
+	{
+		// Don't call the client send message on every Update,
+		// this is sample on how to call the client,
+		// not an example on how to code.
+		if (Input.GetKeyDown(KeyCode.Space))
+		{
+			_client.SendWhisper(_channelToConnectTo, "Your mom");
```

Comment:
I do not understand this action. What does the provided message "Your mom" give in terms of information?

Reply:
This was a temporary debug message this will be changed